### DISCUSSION TITLE
adding devices

### DISCUSSION THREAD

**Discussion by HannesBulk** [SpenceKonde/megaTinyCore#374] [Future Development Discussion]
The MegaTinyCore supports all new AvrTiny and AvrDX processors, however the Mega 0 series (AVR4809 family) are missing. The protocol for programming these devices is identical to programming the ATtiny 0,1 and 2 processors, the only difference is that Flash starts at 0x4000 instead of 0x8000. It would be nice to add these processors, this makes it possible to program these devices with the serial port method, using a single resistor.
One additional remark, instead of using a resistor, is is better to use a simple signal diode, anode to UPDI, Cathode to  TX pin of UPDI-serial converter or. The processor has a build-in pull-up in the UPDI pin. Using a resistor also implies that the resistor acts as a pull-up resistor during when receiving date from the processor, the diode doesn't. This is also the reason why in the Jtag2UPDI program there is no pull-up on the programming pin, it is better to use the internal pull-up on the processor. I tested this, and didn,t have any problems in programming the chips with Baudrates up to 600.000 bps, so quite fast.

**Comment by SpenceKonde** [comment]
I already have my hands full with the cores I am maintaining already. I recommend MegaCoreX  by MCUDude, he does much the same thing I do - only for every part with ATmega in the name. We routinely exchange code and information relating to core development (though he has far more deference towards the official Arduino cores and design decisions than I). He is the one who writes the wrapper libraries for event/logic/etc. I think he is uncomfortable with the pymcuprog tool that we use because it is most certainly not open source (the pymcuprog tool we based it on (all the microchip people on the zoom call were HORRIFIED at the thought that we would be using pyupdi instead of pymcuprog and argued vigorously for the maintained and regression tested pymcuprog instead. 

But no, I do NOT have time to maintain any more cores!

Where do you claim this diode should be used? Are you saying to replace the pyupdi 4.7k resistor with... a fast-switching signal diode? I suppose that should work, yeah, and gets around the fact that the 4.7k value has to thread a needle 0 due to the miserable pin driver strength on the reset/UPDI pin of the tinyAVR parts (it's like that because they have to be able to take the 12v HV programming signal,, and that makes it hard to have good output drivers. But I remember calculating out the range of values that would work and finding it alarmingly narrow. I'll have to try that

**Reply by HannesBulk** [reply]
The Pyupdi with resistor works with the processors in your libraries, however when selecting the MegacoreX from MCUDude, this programmer is not available in the Arduino. I Guess is's better to ask MCUDude to add this programming method in his project.
Thnks, Hannes B

**Comment by SpenceKonde** [comment]
Oh - someone else suggested the same thing just today on DxCore discussion (though they, ah, went in to [a bit more detail](http://ad7i.net/main/analysis-of-resistor-values-for-pymcuprog-usbusart-serial-programming/) - having some kind of resistance between the two pins is important, you need to prevent excessive current flow if the two parts end up completely out of sync somehow, and try to drive it in opposite directions, On the tinyAVR parts, you might be okay, but on the AVR DD parts, for example? Not so much.
